src/cst/NodeSelector.py

```diff
@@ -12,7 +12,7 @@
     PandasInput(),
 ]
 
-module_by_name = {module.module_name: module for module in input_modules}
+module_by_name: dict[str, InputModule] = {module.module_name: module for module in input_modules}
```

Comment:
This typing is probably not necessary if input_modules and the method module_name are typed. You could check this by removing the typing and hovering over the variable and you will see the inferred data type :)

Reply:
yep. actually this is the reason i added the explicit type hint. The code analysis was smart enough to notice that input_modules only contains a PandasInput which is a more concrete interface than what I wanted to get displayed in my code completion :D
But I notice that I alternatively could have changed the type hint of input_modules as well to avoid this problem.
Or add a DaskInput o.o